srcpkgs/mellowplayer/template

```diff
@@ -1,22 +1,23 @@
 # Template file for 'mellowplayer'
 pkgname=mellowplayer
-version=3.3.5
+version=3.4.0
 revision=1
 wrksrc=MellowPlayer-${version}
-build_style=cmake
-hostmakedepends="pkg-config qt5-tools qt5-qmake"
```

Comment:
you need `qbs` and `qt5-qmake` either here in hostmakedepends or add them in `common/environment/build-style/qbs-qt5.sh` Also I would be surprised if the qbs-qt5 build-style actually cross compiles.

Reply:
Yup, cross compiling doesn't work as of now - but QBS docs on cross compiling seem a little "suboptimal" to me (or I just don't find the right ones), so I'm not quite sure how I would get it to cross compile.

Reply:
Adding a cross toolchain should be fairly straight forward e.g. `qbs setup-toolchains aarch64-linux-gnu-gcc aarch64 `. The biggest problem is `qmake` or better `qbs-setup-qt /usr/lib/qt5/bin/qmake qt5` since qmake ( `qmake-qt5 -query` ) will always return the host configuration (which is AFAIK also a problem in qmake build-style). In my understanding of qmake (or lack) we would need to compile qmake for every target architecture we want to cross-build for to feed qbs with the correct path, but I highly suspect that this would break other stuff.

TL;DR qmake sucks and qbs depends on it.

Reply:
> Adding a cross toolchain should be fairly straight forward e.g. qbs setup-toolchains aarch64-linux-gnu-gcc aarch64

Yup, I've done that.

> The biggest problem is qmake or better qbs-setup-qt /usr/lib/qt5/bin/qmake qt5 since qmake ( qmake-qt5 -query ) will always return the host configuration (which is AFAIK also a problem in qmake build-style).

If I understand correctly qbs offers a `sysroot` option (and so does qmake), so if we were to set `sysroot` to /usr/$XBPS_CROSS_TRIPLET it may work. 

> In my understanding of qmake (or lack) we would need to compile qmake for every target architecture we want to cross-build for to feed qbs with the correct path

Uhh..how would we run that cross compiled qmake on the host? qbs runs `qmake -query` to get Qt's paths.

> TL;DR qmake sucks and qbs depends on it.

Agreed.

Reply:
> One can always compile it natively.

Sure, I like my package manager to do that though and to upstream my work, so I guess I'm stuck with this

Reply:
Last I heard, QT is not planning switching build systems until QT 6, which will first appear in late 2020.

> > One can always compile it natively.
> 
> Sure, but I like my package manager to do that though and to upstream my work, so I guess I'm stuck with this

Not ideal, but there seems to be an appimage available for the latest mellow player.

BTW why did mellow player guy switch cold turkey to a build system that no one uses? At the very least, he could of kept the qmake option.

Reply:
> Not ideal, but there seems to be an appimage available for the latest mellow player.

Yeah, not a huge fan of outdated libs :)

>  BTW why did mellow player guy switch cold turkey to a build system that no one uses? At the very least, he could of kept the qmake option.

Uhh...MellowPlayer never offered a QMake option, previously it used CMake for building. But apparently it's easier for the MellowPlayer maintainer to use qbs because it's a little nicer to use for Qt projects than CMake (but hell for everyone else)

Reply:
err right.. in any case, cmake is a lot nicer than qmake or qbs from the end-user perspective.

But who knows... maybe QT will adopt meson and deprecate qbs

Reply:
Yep, CMake (or meson!) would be way nicer, but oh well. I very much doubt that Qt won't use qbs, why would they write an entire build system only to not use it? This might not be the right place to discuss this though